ec2/sc-portfolio-ec2-development.yaml

```diff
@@ -0,0 +1,152 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: '[DEVELOPMENT] EC2 Portfolio for Service Catalog.'
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Portfolio Information
+        Parameters:
+          - PorfolioName
+          - PortfolioProvider
+          - PorfolioDescription
+      - Label:
+          default: IAM Settings
+        Parameters:
+          - LaunchRoleName
+          - PrincipalRoleName1
+          - PrincipalGroupName1
+          - PrincipalRoleName2
+          - PrincipalGroupName2
+      - Label:
+          default: Product Settings
+        Parameters:
+          - RepoRootURL
+Parameters:
+  PortfolioProvider:
+    Type: String
+    Description: Provider Name
+    Default: IT Services
+  PorfolioName:
+    Type: String
+    Description: Portfolio Name
+    Default: '[DEVELOPMENT] Service Catalog EC2 Reference Architecture'
+  PorfolioDescription:
+    Type: String
+    Description: Portfolio Description
+    Default: Service Catalog Portfolio that contains reference architecture
+      products for Amazon Elastic Compute Cloud (EC2).
+  LaunchRoleName:
+    Type: String
+    Description: Name of the launch constraint role for EC2 products. leave this
+      blank to create the role.
+  PrincipalRoleName1:
+    Type: String
+    Description: (Optional) Name of an IAM role which can execute products
+      in this portfolio.
+    Default: ''
+  PrincipalGroupName1:
+    Type: String
+    Description: (Optional) Name of an IAM  group which can execute products
+      in this portfolio.
+    Default: ''
+  PrincipalRoleName2:
```

Comment:
why do we need a 2nd principal role? and why are both optional?  maybe update description or add a comment with explanation would help.

Reply:
Re alternate principal role/group, I was following the pattern that previously existed in the file where there were alternate roles. I can see that there might be a need for an alternate role/group in the future.

Why they are all optional: I don't want that to be the case, but which one am I going to require? The role, the group, or both?

Reply:
it seems like sc-portfolio-ec2.yaml will create a user/role/group if one isn't specified and maybe that's why it's optional?   It seems like you'll need at least one group and role and the group and role needs to be created before executing this template.  That most likely means that we need a separate base template to create these groups and roles.  We would need to execute that base template before making this template available in the SC.  do we have such a base template?